src/main/java/org/candlepin/subscriptions/tally/InventorySwatchDataCollator.java

```diff
@@ -91,10 +91,15 @@ public InventorySwatchDataCollator(
    */
   public int collateData(String orgId, int culledOffsetDays, Processor processor) {
     Stream<InventoryHostFacts> inventorySystemStream =
-        inventoryRepository.streamFacts(orgId, culledOffsetDays);
+        inventoryRepository
+            .streamFacts(orgId, culledOffsetDays)
+            .sorted(Comparator.comparing(SortKey::fromHbiSystem));
     Stream<String> activeSubmanIdStream =
         inventoryRepository.streamActiveSubscriptionManagerIds(orgId, culledOffsetDays);
-    Stream<Host> swatchSystemStream = hostRepository.streamHbiHostsByOrgId(orgId);
+    Stream<Host> swatchSystemStream =
+        hostRepository
+            .streamHbiHostsByOrgId(orgId)
+            .sorted(Comparator.comparing(SortKey::fromSwatchSystem));
```

Comment:
This should be a no-op given the order-by clause:
https://github.com/RedHatInsights/rhsm-subscriptions/blob/3fb6479130cd1118346f1065f396e6b235df3039/swatch-core/src/main/java/org/candlepin/subscriptions/db/HostRepository.java#L73

should be equivalent to the `SortKey`'s `compareTo` implementation:

https://github.com/RedHatInsights/rhsm-subscriptions/blob/9050d6340388cc41ee15c3f771cd96ab15519555/src/main/java/org/candlepin/subscriptions/tally/InventorySwatchDataCollator.java#L268-L281